protocol/lib/log/utils.go

```diff
@@ -0,0 +1,37 @@
+package log
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// InfoLog reports msg as an info level log with specified key vals.
+func InfoLog(ctx sdk.Context, msg string, keyvals ...interface{}) {
+	ctx.Logger().Info(msg, keyvals...)
+}
+
+// DebugLog reports msg as a debug level log with specified key vals.
+func DebugLog(ctx sdk.Context, msg string, keyvals ...interface{}) {
+	ctx.Logger().Debug(msg, keyvals...)
+}
+
+// ErrorLogWithError reports msg as a error log with specified key vals,
+// as well as attaching the error object to the log for datadog error tracking.
+func ErrorLogWithError(ctx sdk.Context, msg string, err error, keyvals ...interface{}) {
+	ctx.Logger().Error(msg, append(keyvals, Error, err))
+}
+
+// ErrorLog reports msg as a error log. It constructs an error object on the fly with
+// the given message object.
+// Please try to use define a new message and use `ErrorLogWithError` instead.
```

Comment:
Assuming u meant this

```suggestion
// Please try to define a new message and use `ErrorLogWithError` instead.
```